src/pages/patientView/vafPlot/VAFPlot.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import $ from 'jquery';
+import {AlleleFreqPlotMulti} from './legacyVAFCode.js';
+
+export type IVAFPlotProps = {
+    data?: { [s: string]:number[] };
+    order?: { [s:string]:number };
+    colors?: { [s: string]:string };
+    labels?: { [s:string]:string };
+    width?: number;
+    height?: number;
+    margin_left?: number;
+    margin_right?: number;
+    margin_top?: number;
+    margin_bottom?: number;
+    xticks?: number;
+    yticks?: number;
+    label_font_size?: string;
+    nolegend?: boolean;
+    init_show_histogram?: boolean;
+    init_show_curve?: boolean;
+    show_controls?: boolean;
+};
+export class VAFPlot extends React.Component<IVAFPlotProps, {}> {
+
+    constructor(props) {
+        super(props)
+
+        // Compute contingent default params
+        let label_dist_to_axis = (this.props.xticks === 0) ? 13: 30;
+
+        let options = {
+            label_font_size: this.props.label_font_size,
+            xticks: this.props.xticks,
+            yticks: this.props.yticks,
+            nolegend: this.props.nolegend,
+            margin: {
+                top: this.props.margin_top,
+                left: this.props.margin_left,
+                right: this.props.margin_right,
+                bottom: this.props.margin_bottom
+            },
+            width: this.props.width,
+            height: this.props.height
+        };
+        // margin bottom must be computed before height because height computation makes use of it
+        options.margin.bottom = (isNaN(options.margin.bottom) ? (30 + (label_dist_to_axis / 2)) : options.margin.bottom);
```

Comment:
these are used here actually, it's one of the ways we set up the thumbnail and the expanded version to be different.

But I've replaced isNaN with an undefined check.